java/src/main/native/src/TableJni.cpp

```diff
@@ -1110,56 +1112,101 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_readParquet(
   CATCH_STD(env, NULL);
 }
 
+int set_column_metadata(cudf::io::column_in_metadata &column_metadata,
+                         std::vector<std::string> &col_names, 
+                         cudf::jni::native_jbooleanArray &nullability,
+                         cudf::jni::native_jbooleanArray &isInt96,
+                         cudf::jni::native_jintArray &precisions,
+                         cudf::jni::native_jintArray &children, int read_index) {
+  int write_index = 0;
+  int num_children = children[read_index++];
```

Comment:
I found it confusing that we're _not_ reading the names, etc. from the specified read_index but actually read_index + 1 because of this.  It might be clearer if `num_children` is passed into this method since every caller is already loading it to decide whether to call this, and then the algorithm can start reading names, etc. from `read_index` as expected.